### ISSUE TITLE
Cannot retrieve the date of last acquisition

### ISSUE THREAD

**Issue by dgateau** [Yadoms/yadoms#532]
When getting value from database (retrieving sensor value or virtual equipment state) in a Python script for example, it is not possible to get the date of the last acquisition/update for this value. for example, it's not possible to determine the lost of communication with an equipment from a script...

**Comment by sgallou** [comment]
We will add a reakKeywordDetails function.
But to achieve your need, you can look at this rule, who should do the job :

```python
import scriptUtilities
import time
import datetime


def yMain(yApi):
   print ('Script started')
   
   doorKeywordId = yApi.getKeywordId('Ouverture porte d\'entrée', 'CO - Contact')
   print 'doorKeywordId =', doorKeywordId
   lastValueDateTime = datetime.datetime.now()
   deviceIsLost = False
   
   while True:
      newValue = yApi.waitForNextAcquisition(doorKeywordId, "" if deviceIsLost else "00:30:00")
      if not newValue:
         deviceIsLost = True
         print '! ! ! Device disparu (aucun signal reçu depuis plus de 30 minutes) ! ! !'
      else:
         deviceIsLost = False
         newValueDateTime = datetime.datetime.now()
         elapsedSeconds = (newValueDateTime - lastValueDateTime).seconds
         elapsedMinutes = elapsedSeconds / 60
         lastValueDateTime = newValueDateTime
         print ('Fermée ' if newValue == '1' else 'Ouverte'), '(délai depuis dernier événement =', (elapsedMinutes if elapsedMinutes > 2 else elapsedSeconds), (' minutes)' if elapsedMinutes > 2 else ' secondes)')
```

**Comment by dgateau** [comment]
First, thank you for your great job on Yadoms, and for your fast answers and support.
In my case, I'm not using "waitForNextAcquisition" loop, since i'm waiting for user interaction on MMI. So I have chosen to make polling... And doing so, I cannot timestamp my acquisitions.
But (while waiting for the "readKeywordDetails " function to be released), I will try to change my script structure to handle the solution you are suggesting :-)
Thanks!

**Comment by sgallou** [comment]
Closed by dab9873672d5eb3ef388b849132c53978cdba47d
